### PR TITLE
ndctl: Add lightbits infrastructure for building and installing ndctl.

### PR DESCRIPTION
**Pull request by oren-lb**
@sashas-lb please review

### CODE REVIEW

**Comment by sashas-lb** [comment]
oren please split ignores and lightbits build into separate commits

File: lb.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+ndctl:
+  nvdimm-util:
```
Comment (sashas-lb) [inline lb.yaml]: why not calling the baby by name "ndctl" :)

File: Makefile.lb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+install:
+	$(Q)$(MAKE) install
+
+checkin:
+	$(Q)component-tool checkin --repo=ndctl --type=$(BUILD_TYPE) nvdimm-util
+
+clean:
+	$(Q)$(MAKE) clean
+
+.PHONY: install checkin
```
Comment (sashas-lb) [inline Makefile.lb]: also clean and build are phony targets

File: Makefile.lb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+NVDIMM_UTIL_INSTALL_DIR ?= $(shell component-tool localpath --repo=ndctl --type=$(BUILD_TYPE) nvdimm-util)
+
+NDCTL_DIR := $(WORKSPACE_TOP)/ndctl/
+
+build:
+	cd $(NDCTL_DIR) && ./autogen.sh
+	cd $(NDCTL_DIR) && ./configure CFLAGS='-g -O0' --prefix=$(NVDIMM_UTIL_INSTALL_DIR) --sysconfdir=/etc --libdir=$(NVDIMM_UTIL_INSTALL_DIR)/lib
+	$(Q)$(MAKE)
+	$(Q)$(MAKE) check
```
Comment (sashas-lb) [inline Makefile.lb]: what is make check ?

**Review by sashas-lb — COMMENTED** [review]
great job see some small comments inside

File: Makefile.lb (hunk 3 above)
Comment (lital-lb) [reply Makefile.lb]: If this is needed, maybe add a link to the original build sequence instructions in a comment

**Comment by lital-lb** [comment]
Looks good, just make sure 'lb-build' runs as in and builds it well
```
cd $WORKSPACE_TOP/ndctl
dockerize lb-build -o
```

File: Makefile.lb (hunk 3 above)
Comment (oren-lb) [reply Makefile.lb]: yes this is the instructions for the build process for ndctl

File: Makefile.lb (hunk 2 above)
Comment (oren-lb) [reply Makefile.lb]: ok

File: .gitignore [hunk 4]
```diff
@@ -12,3 +12,79 @@ Makefile.in
 /configure
 /libtool
 /stamp-h1
+Documentation/daxctl/asciidoc.conf
```
Comment (maor-lb) [inline .gitignore:15]: Please split gitignore and makefile changes to different commits

File: Makefile.lb (hunk 3 above)
Comment (maor-lb) [reply Makefile.lb]: It's running some kind of unit tests, @oren-lb IMO we can skip this unless we plan to make changes is ndctl

File: lb.yaml [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+    build:
+      - make -f Makefile.lb
```
Comment (maor-lb) [inline lb.yaml]: @lital-lb this a bit cumbersome sequence and return in every repo, why didn't we added default target for build and install ?

File: .gitignore (hunk 4 above)
Comment (oren-lb) [reply .gitignore:15]: Done

File: lb.yaml (hunk 1 above)
Comment (oren-lb) [reply lb.yaml]: done

File: Makefile.lb (hunk 2 above)
Comment (oren-lb) [reply Makefile.lb]: done

File: Makefile.lb (hunk 3 above)
Comment (oren-lb) [reply Makefile.lb]: I prefer to leave it it part of the instructions suggested by the building instructions and it does not add any significant compilation time.
